src/devices/slacky_diy.ts

```diff
@@ -0,0 +1,1804 @@
+import {Zcl} from "zigbee-herdsman";
+
+import * as fz from "../converters/fromZigbee";
+import * as tz from "../converters/toZigbee";
+import * as constants from "../lib/constants";
+import * as exposes from "../lib/exposes";
+import {logger} from "../lib/logger";
+import * as m from "../lib/modernExtend";
+import * as reporting from "../lib/reporting";
+import * as globalStore from "../lib/store";
+import type {
+    Configure,
+    Definition,
+    DefinitionWithExtend,
+    Expose,
+    Fz,
+    KeyValue,
+    KeyValueAny,
+    KeyValueNumberString,
+    ModernExtend,
+    Tz,
+    Zh,
+} from "../lib/types";
+import * as utils from "../lib/utils";
+import {assertString, getFromLookup, getOptions, postfixWithEndpointName, precisionRound} from "../lib/utils";
+
+const e = exposes.presets;
+const ea = exposes.access;
+
+const defaultReporting = {min: 0, max: 300, change: 0};
+const co2Reporting = {min: 10, max: 300, change: 0.000001};
+const batteryReporting = {min: 3600, max: 0, change: 0};
+
+const model_r01 = "Tuya_Thermostat_r01";
+const model_r02 = "Tuya_Thermostat_r02";
+const model_r03 = "Tuya_Thermostat_r03";
+const model_r04 = "Tuya_Thermostat_r04";
+const model_r05 = "Tuya_Thermostat_r05";
+const model_r06 = "Tuya_Thermostat_r06";
+const model_r07 = "Tuya_Thermostat_r07";
+const model_r08 = "Tuya_Thermostat_r08";
+
+const attrThermSensorUser = 0xf000;
+const attrThermFrostProtect = 0xf001;
+const attrThermHeatProtect = 0xf002;
+const attrThermEcoMode = 0xf003;
+const attrThermEcoModeHeatTemperature = 0xf004;
+const attrThermFrostProtectOnOff = 0xf005;
+const attrThermSettingsReset = 0xf006;
+const attrThermScheduleMode = 0xf007;
+const attrThermSound = 0xf008;
+const attrThermLevel = 0xf009;
+const attrThermInversion = 0xf00a;
+const attrThermEcoModeCoolTemperature = 0xf00b;
+const attrThermExtTemperatureCalibration = 0xf00c;
+const attrFanCtrlControl = 0xf000;
+
+const switchSensorUsed = ["Inner (IN)", "All (AL)", "Outer (OU)"];
+
+const attrElCityMeterModelPreset = 0xf000;
+const attrElCityMeterAddressPreset = 0xf001;
+const attrElCityMeterMeasurementPreset = 0xf002;
+const attrElCityMeterDateRelease = 0xf003;
+const attrElCityMeterModelName = 0xf004;
+const attrElCityMeterPasswordPreset = 0xf005;
+
+const fzLocal = {
+    thermostat_child_lock: {
```

Comment:
This can be removed into `fromZigbee.ts` as it is generic (replace `namron_hvac_user_interface`)